app/models/user.rb

```diff
@@ -224,38 +172,24 @@ def project_unsubscribes
     end
   end
 
-  def project_owner?
-    projects.present?
-  end
-
   def fix_twitter_user
     self.twitter.gsub!(/@/, '') if self.twitter
   end
 
   def fix_facebook_link
-    if self.facebook_link.present?
-      self.facebook_link = ('http://' + self.facebook_link) unless self.facebook_link[/^https?:\/\//]
+    if self.facebook_link && !self.facebook_link[/^https?:\/\//]
+      self.facebook_link = ('http://' + self.facebook_link)
     end
   end
 
-  def password_required?
-    !persisted? || !password.nil? || !password_confirmation.nil?
-  end
-
-  def password_confirmation_required?
-    !new_record?
-  end
-
   def has_valid_contribution_for_project?(project_id)
-    contributions.with_state(['confirmed', 'requested_refund', 'waiting_confirmation']).where(project_id: project_id).present?
-  end
-
-  def generate_reset_password_token
-    raw, enc = Devise.token_generator.generate(self.class, :reset_password_token)
-    self.reset_password_token   = enc
-    self.reset_password_sent_at = Time.now.utc
-    self.save(validate: false)
-    raw
+    contributions.with_state(
+      [
+        'confirmed',
+        'requested_refund',
+        'waiting_confirmation'
+      ]
+    ).where(project_id: project_id).present?
```

Comment:
use o exist? nesse caso ai o AR ja faz o trabalho pra vc http://guides.rubyonrails.org/active_record_querying.html#existence-of-objects